internal/wasm/jit/engine.go

```diff
@@ -450,22 +450,25 @@ const (
 
 // execHostFunction executes the given host function represented as *reflect.Value.
 //
-// The arguments to the function are popped from the stack stack following the convension of
-// Wasm stack machine.
+// The arguments to the function are popped from the stack following the convention of the Wasm stack machine.
 // For example, if the host function F requires the (x1 uint32, x2 float32) parameters, and
 // the stack is [..., A, B], then the function is called as F(A, B) where A and B are interpreted
 // as uint32 and float32 respectively.
 //
 // After the execution, the result of host function is pushed onto the stack.
 //
 // ctx parameter is passed to the host function as a first argument.
-func (e *engine) execHostFunction(f *reflect.Value, ctx *wasm.HostFunctionCallContext) {
+func (e *engine) execHostFunction(fk wasm.FunctionKind, f *reflect.Value, ctx *wasm.HostFunctionCallContext) {
 	tp := f.Type()
 	in := make([]reflect.Value, tp.NumIn())
 
 	// We pop the value and pass them as arguments in a reverse order according to the
 	// stack machine convention.
-	for i := len(in) - 1; i >= 1; i-- {
+	wasmParamOffset := 0
+	if fk != wasm.FunctionKindHost {
+		wasmParamOffset = 1
+	}
+	for i := len(in) - 1; i >= wasmParamOffset; i-- {
```

Comment:
so this and the `GetHostFunctionCallContextValue` below are the additional codes on the host call's runtime pass right? TODO comment would be appreciated for future performance optimizations

Reply:
so the fk is compared twice.  here to determine if there is an offset or not, and inside GetHostFunctionCallContextValue to determine what to set. so this is read twice vs the code that existed before and if reading a number twice is a perf hit we are concerned with we can optimize!

Reply:
I made a general note instead as I think this was too specific, and can be handled in a broader scope inside that same function and ideally get better perf regardless of context. we'll see..